DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -0,0 +1,166 @@
+                        true
+                    )
+                );
+            }
+        }
+
+        [Test]
+        public void StartOrRestartSession_Should_Start_Subsequent_Session()
+        {
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 101;
+                const int customisationId = 1240;
+                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
+
+                // Then
+                var newSession = sessionTestHelper.GetSession(sessionId);
+
+                newSession.Should().NotBeNull();
+                SessionTestHelper.SessionsShouldBeApproximatelyEquivalent(
+                    newSession!,
+                    new Session(
+                        sessionId,
+                        candidateId,
+                        customisationId,
+                        DateTime.Now,
+                        0,
+                        true
+                    )
+                );
+            }
+        }
+
+        [Test]
+        public void StartOrRestartSession_Should_Stop_Other_Sessions()
+        {
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 101;
+                const int customisationId = 1240;
+                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
+
+                // Then
+                var sessions = sessionTestHelper.GetCandidateSessions(candidateId);
+                var activeSessions = sessions.Where(session => session.Active);
+
+                activeSessions.Should().HaveCount(1);
+                activeSessions.First().SessionId.Should().Be(sessionId);
+            }
+        }
+
+        [Test]
+        public void StopSession_Should_Stop_Candidates_Sessions()
+        {
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 101;
+                sessionService.StopSession(candidateId);
+
+                // Then
+                var sessions = sessionTestHelper.GetCandidateSessions(candidateId);
+                var activeSessions = sessions.Where(session => session.Active);
+
+                activeSessions.Should().BeEmpty();
+            }
+        }
+
+        [Test]
+        public void UpdateSessionDuration_Should_Only_Update_Given_Active_Sessions()
+        {
+            const int twoMinutesInMilliseconds = 120 * 1000;
+
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 9;
+                const int sessionId = 473;
+                var startingSessions = sessionTestHelper.GetCandidateSessions(candidateId);
+
+                sessionService.UpdateSessionDuration(sessionId);
+
+                // Then
+                var newSessions = sessionTestHelper.GetCandidateSessions(candidateId).ToList();

```

Comment:
I think `updatedSessions` might be more descriptive? As otherwise it's easy to mistake these for sessions which have just been added 🙂

Reply:
Yeah I agree. Changed in a4825ca 🙂